File: src/main/java/github/lms/approval/infra/ApprovalActionLogRepository.java
package github.lms.approval.infra;

import github.lms.approval.domain.ApprovalActionLog;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface ApprovalActionLogRepository extends JpaRepository<ApprovalActionLog, UUID> {
    Optional<ApprovalActionLog> findByApprovalIdAndStepIdAndApproverIdAndIdempotencyKey(
            UUID approvalId, UUID stepId, UUID approverId, String idempotencyKey);
}

File: src/main/kotlin/github/lms/approval/infra/ApprovalActionLogRepository.kt
package github.lms.approval.infra

import github.lms.approval.domain.ApprovalActionLog
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ApprovalActionLogRepository : JpaRepository<ApprovalActionLog, UUID> {
    fun findByApprovalIdAndStepIdAndApproverIdAndIdempotencyKey(
        approvalId: UUID,
        stepId: UUID,
        approverId: UUID,
        idempotencyKey: String
    ): ApprovalActionLog?
}
